docker-start.sh

```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.35.3/install.sh | bash
+source ~/.bashrc
+
+nvm install 12.16.1
+nvm use --delete-prefix 12.16.1
+npm install -g envsub
+
+cd /app/website
+# grab Helm values from env and update our .env file
+# Docusaurus build process means we can't just grab the process.env.$var
+# We have to update our .env file which will be read by a Docusaurus plugin (dotenv plugin)
+
+envsub .env.docker .env
```

Comment:
> lol so docusaurus requires that env is provided through `.env`? running something like `ALGOLIA_API_KEY=somekey npm run build` doesn't work eh?

Yeah I didn't have any luck passing a var straight to the process in the build script.



> also if `envsub` is already in `package.json`, does `npm ci` and then `npx envsub ...` work? that way we can skip the global install of `envsub` and have the version locked down

great idea

Reply:
> lol so docusaurus requires that env is provided through `.env`? running something like `ALGOLIA_API_KEY=somekey npm run build` doesn't work eh?

Scratch my earlier comment. We can pass as suggested. Is this preferable? @hlminh2000 
Downside is we're manually updating the `docker-start.sh`, this way we can just update Helm values and our local `env.docker` file

Reply:
if the process is able to pick up the env without `.env`, there's no need to define the variables here at all, because kubernetes will have the variables defined in the process. It's more important to make sure the application code has a centralized place where env is being read and not just `process.env.SOMEVAR` scattered around (not a devops problem though) :)

Reply:
@hlminh2000 right right right! got it now. All the vars will be in the global scope for the script, Thanks!